front/app/containers/Admin/reporting/containers/ReportBuilder/index.tsx

```diff
@@ -42,6 +42,23 @@ interface Props {
   reportLayout: ReportLayout;
 }
 
+const areEqual = (
+  prevData: Record<string, object>,
+  nextData: Record<string, object>
+) => {
+  // when we save the data, `undefined` field values are not saved
+  // (e.g., `parent` in { ROOT: { parent: undefined, ...}, ...}).
+  // So, they are never present in reportLayout.attributes.craftjs_json,
+  // but they are present in the editor state `query.getSerializedNodes()`.
+  // JSON.stringify emulates sending the data to the server and getting it back
+  // (it removes undefined values).
+  // JSON.parse makes sure that the comparison is not affected by the order of keys.
+  return isEqual(
+    JSON.parse(JSON.stringify(prevData)),
```

Comment:
In theory, may be not performant, but I just tested it with some production report and it actually works quite fast.
```
console.time('parse-stringify'); JSON.parse(JSON.stringify(obj)); console.timeEnd('parse-stringify')
```
No throttling: `parse-stringify: 0.193115234375 ms`
Low-end mobile: `parse-stringify: 1.547119140625 ms`